static/skin/index.js

```diff
@@ -429,7 +415,11 @@
             filter.addEventListener('change', () => {resetAndFilter(filter.name, filter.value)});
         });
         if (filters) {
-            window.history.pushState({}, null, `?${params.toString()}`);
+            const currentLink = window.location.search;
+            const newLink = `?${params.toString()}`;
+            if (currentLink != newLink) {
+                window.history.pushState({}, null, newLink);       
+            } 
```

Comment:
Please elaborate the description of the commit with the negative side-effects of `window.history.pushState()` being called unconditionally (the worst of which is the discarded forward history when the current page is loaded as a result of navigating back).

Reply:
Reading the updated commit message I see that you don't fully realize the problem with the unconditional `window.history.pushState()` here. It is independent of the handling of the "reset filter" link as demonstrated in https://github.com/kiwix/libkiwix/pull/711#discussion_r905029154. Therefore I recommend that this commit is also split into two changes:

1. Fix the problem with `window.history.pushState()` in `window.onload` with a proper explanation (you can include the example from https://github.com/kiwix/libkiwix/pull/711#discussion_r905029154).
2. Drop the `onclick` handler of the "reset filter" link noting that it was a workaround of a more general issue now fixed by the previous commit.

Reply:
Correct, thank you.